### PR TITLE
Yaml update cloud build subs

### PR DESCRIPTION
**Pull request by goog-lukemc**
Updated the yaml file to allow for easier work with testing - also updated the docker file for cloud run.

### CODE REVIEW

File: gcp-data-drive/cloudbuild_appengine.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 steps:
 - name: 'gcr.io/cloud-builders/git'
-  args: ['clone','https://github.com/GoogleCloudPlatform/DIY-Tools.git']
+  args: ['clone','--single-branch','--branch','${_GIT_SOURCE_BRANCH}','${_GIT_SOURCE_URL}']
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_appengine.yaml]: Why do you think you need to do this?  Isn't the code already present?

File: gcp-data-drive/cloudbuild_gcf.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 steps:
 - name: 'gcr.io/cloud-builders/git'
-  args: ['clone','https://github.com/GoogleCloudPlatform/DIY-Tools.git']
+  args: ['clone','--single-branch','--branch','${_GIT_SOURCE_BRANCH}','${_GIT_SOURCE_URL}']
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_gcf.yaml]: Why do you think you need to do this?  Isn't the code already present?

File: gcp-data-drive/cloudbuild_run.yaml [hunk 3]
```diff
@@ -12,19 +12,17 @@
...
-# - name: 'golang:1.13'
-#   args: ['go','build','-mod=readonly','-v','-o','gcp-data-drive']
-#   env: ['GO111MODULE=on','CGO_ENABLED=0','GOOS=linux','GOARCH=amd64']
-#   dir: 'DIY-Tools/gcp-data-drive/cmd/webserver'
-#
-# - name: 'gcr.io/cloud-builders/gcloud'
-#   args: ['builds','submit','--tag','gcr.io/$PROJECT_ID/gcp-data-drive']
-#   dir: 'DIY-Tools/gcp-data-drive'
+# Switch clone from https://github.com/SureSkills/DIY-Tools.git to
+# https://github.com/GoogleCloudPlatform/DIY-Tools
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_run.yaml]: explain why?

File: gcp-data-drive/cloudbuild_run.yaml [hunk 4]
```diff
@@ -12,19 +12,17 @@
...
+
+steps:
+- name: 'gcr.io/cloud-builders/git'
+  args: ['clone','--single-branch','--branch','${_GIT_SOURCE_BRANCH}','${_GIT_SOURCE_URL}']
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_run.yaml]: Should be present already.

File: gcp-data-drive/cloudbuild_run.yaml [hunk 5]
```diff
@@ -12,19 +12,17 @@
...
+
+- name: 'gcr.io/cloud-builders/gcloud'
+  args: ['builds','submit','--tag','gcr.io/$PROJECT_ID/gcp-data-drive']
+  dir: 'DIY-Tools/gcp-data-drive'
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_run.yaml]: You are trying to chain cloud build steps?  Huh?
Why not have the same trigger start several steps?

File: gcp-data-drive/cloudbuild_run.yaml [hunk 6]
```diff
@@ -12,19 +12,17 @@
...
+
+- name: 'gcr.io/cloud-builders/gcloud'
+  args: ['run','deploy','gcp-data-drive','--image','gcr.io/$PROJECT_ID/gcp-data-drive','--platform','managed','--region','us-central1','--allow-unauthenticated']
+  dir: 'DIY-Tools/gcp-data-drive'
```
Comment (lesv) [inline gcp-data-drive/cloudbuild_run.yaml]: Including this one?

File: gcp-data-drive/Dockerfile [hunk 7]
```diff
@@ -19,14 +19,25 @@
...
+COPY go.* ./
+RUN go mod download
 
-FROM gcr.io/distroless/static
+# Copy local code to the container image.
+COPY .  ./
+
+WORKDIR /app/cmd/webserver
+# Build the binary.
+RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -mod=readonly -v -o gcp-data-drive
```
Comment (lesv) [inline gcp-data-drive/Dockerfile]: What does -mod=readonly do?

File: gcp-data-drive/cloudbuild_appengine.yaml (hunk 1 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_appengine.yaml]: Added comments to the file per our conversation.

File: gcp-data-drive/cloudbuild_gcf.yaml (hunk 2 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_gcf.yaml]: Added comments to the file per our conversation.

File: gcp-data-drive/cloudbuild_run.yaml (hunk 3 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_run.yaml]: Removed.  It was an oversite on my part.

File: gcp-data-drive/cloudbuild_run.yaml (hunk 4 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_run.yaml]: Added comments to the file per our conversation.

File: gcp-data-drive/cloudbuild_run.yaml (hunk 5 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_run.yaml]: As we discussed.  It is the new default for the cloud build command.

File: gcp-data-drive/Dockerfile (hunk 7 above)
Comment (goog-lukemc) [reply gcp-data-drive/Dockerfile]: Added comments to the file to explain. The TLDR is - It forces the build to fail if there any an issue with the go.mod that can be auto resolved by auto editing the go.mod.

File: gcp-data-drive/cloudbuild_run.yaml (hunk 6 above)
Comment (goog-lukemc) [reply gcp-data-drive/cloudbuild_run.yaml]: Added comments to the file per our conversation.

**Pull request merged**